### PR TITLE
FEAT: Add Filter Management

### PR DESCRIPTION
**Pull request by VishalPatil18**
### 📝 Description

With filters, the user could filter out products of their choice on the product listing page.

### 🚀 New behavior
The user would be able to
-  filter on the basis of price range
- sort the products low-to-high or high-to-low according to price
- filter out products related to specific categories
- the `buy button` and `wishlist button` state is now persisting on-screen change
- the `cart` and `wishlist` data is persisting on logout and login again

### 🖼 Live Preview

https://viskart-react-version-dex09glng-vishalpatil18.vercel.app/

### 💣 Is this a breaking change?
- [x] Yes - (The new games section on the landing page and top products on the landing page is breaking. Hence, I've commented that as of now. I will fix that in the next PR)
- [ ] No

### 🔬 Is the code self-reviewed?
- [x] Yes
- [ ] No

### 📋 Is the code well-formatted?
- [x] Yes
- [ ] No

### 🔗 Related to Issue 
- #14 

> NOTE: The site is not responsive yet. I would work on responsiveness at the end.

### CODE REVIEW

**Review by mohammed-sas — APPROVED** [review]
1.functionalities are working as stated in description
2. utils has been coded good

**Comment by kart-c** [comment]
If I sort products from "low to high"/"high to low", it works as expected. But when I clear the filters, the last sort still persists. Can you check this?

File: src/utilities/categoryFilterHandler.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const categoryFilterHandler = (products, categories) => {
+  return products.filter((product) => {
+    for (let i = 0; i < categories.length; i++) {
```
Comment (kart-c) [inline src/utilities/categoryFilterHandler.js:3]: you can use map here

**Review by kart-c — APPROVED** [review]
Looks good
- Good work on fixing the sorting bug 
- All the filters are working good
- UI is really good

File: src/utilities/categoryFilterHandler.js (hunk 1 above)
Comment (VishalPatil18) [reply src/utilities/categoryFilterHandler.js:3]: Sure will do it in next PR. For the time being I've created an [issue](https://github.com/VishalPatil18/viskart-react-version/issues/16) for the same.

**Pull request merged**